hoodie-client/src/main/java/com/uber/hoodie/HoodieReadClient.java

```diff
@@ -234,9 +238,10 @@ private void assertSqlContext() {
         }
 
         try {
-            HoodieCommitMetadata commitMetdata =
+            HoodieCommitMetadata commitMetadata =
                     HoodieCommitMetadata.fromBytes(commitTimeline.getInstantDetails(commitInstant).get());
-            Collection<String> paths = commitMetdata.getFileIdAndFullPaths().values();
+            Collection<String> relativePaths = commitMetadata.getFileIdAndRelativePaths().values();
+            Collection<String> paths = relativePaths.stream().map(c -> commitMetadata.getCurrentBasePath().toString() + "/" + c).collect(Collectors.toList());
```

Comment:
Since similar logic is used above too, you might want to move this into `HoodieCommitMetadata` itself as before.

Also you might want to use hdfs `Path` object to compute children rather than working with strings directly. (I'll have to check if there are path utilities in hoodie already)